src/main/java/org/openrewrite/staticanalysis/SimplifyBooleanExpressionWithDeMorgan.java

```diff
@@ -0,0 +1,123 @@
+
+    @Override
+    public String getDisplayName() {
+        return "Simplify boolean expressions using De Morgan's laws";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Applies De Morgan's laws to simplify boolean expressions with negation. " +
+                "Transforms `!(a && b)` to `!a || !b` and `!(a || b)` to `!a && !b`.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-1125");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaVisitor<ExecutionContext>() {
+
+            @Override
+            public J visitUnary(J.Unary unary, ExecutionContext ctx) {
+                if (unary.getOperator() == J.Unary.Type.Not && unary.getExpression() instanceof J.Parentheses) {
+                    J.Parentheses<?> parentheses = (J.Parentheses<?>) unary.getExpression();
+                    if (parentheses.getTree() instanceof J.Binary) {
+                        J.Binary binary = (J.Binary) parentheses.getTree();
+                        J.Parentheses<J.Binary> parenthesesBinary = (J.Parentheses<J.Binary>) unary.getExpression();
+
+                        J.Binary.Type newOperator = null;
+                        if (binary.getOperator() == J.Binary.Type.And) {
+                            newOperator = J.Binary.Type.Or;
+                        } else if (binary.getOperator() == J.Binary.Type.Or) {
+                            newOperator = J.Binary.Type.And;
+                        }
+                        Expression left = binary.getLeft();
+                        Expression right = binary.getRight();
+
+                        if (newOperator != null) {
+                            left = negate(left);
+                            left = (Expression) new ParenthesizeVisitor<>().visit(left, ctx);
+                            right = negate(right);
+                            right = (Expression) new ParenthesizeVisitor<>().visit(right, ctx);
+                        }
+
+                        left = (Expression) this.visit(left, ctx);
+                        right = (Expression) this.visit(right, ctx);
+
+                        if (newOperator == null) {
+                            J.Binary visitedBinary = binary.withLeft(left).withRight(right);
+                            return unary.withExpression(parenthesesBinary.withTree(visitedBinary));
+                        }
+                        Space prefix = unary.getPrefix();
+                        List<Comment> comments = new ArrayList<>(prefix.getComments());
+                        comments.addAll(parenthesesBinary.getComments());
+                        comments.addAll(binary.getComments());
+                        prefix = prefix.withComments(comments);
```

Comment:
`ListUtils.concat` might be somewhat preferred here, just to ensure that pattern is used where applicable as well.